src/core/toga/widgets/numberinput.py

```diff
@@ -129,19 +125,20 @@ def value(self):
 
     @value.setter
     def value(self, value):
-        try:
-            self._value = Decimal(value)
-
-            if self.min_value is not None and self._value < self.min_value:
-                self._value = self.min_value
-            elif self.max_value is not None and self._value > self.max_value:
-                self._value = self.max_value
-        except (ValueError, InvalidOperation):
-            raise ValueError("value must be a number")
-        except TypeError:
+        if value is not None:
+            try:
+                self._value = Decimal(str(value))
+
+                if self.min_value is not None and self._value < self.min_value:
+                    self._value = self.min_value
+                elif self.max_value is not None and self._value > self.max_value:
+                    self._value = self.max_value
+            except (ValueError, InvalidOperation):
+                raise ValueError("value must be a number")
+        else:
             self._value = None
 
-        self._impl.set_value(value)
+        self._impl.set_value(self._value)
```

Comment:
At least in principle, the old code is intentional here. The underlying idea here is that the `value` passed in as an argument is the "raw" value; if the widget needs the converted value, it should interrogate `self.interface.value`. There aren't many cases where the raw value is needed, but they exist (or... at least they did at some point in the past...). 

Does this actually impact on the fix you're proposing? Based on your comment on the original ticket, I'm guessing not?